<!-- To be included inside template.html -->

<div class="page-header">
    <h1>Explore</h1>
</div>

This page contains materials for the data processed.
<div>
    <ul>
        <li><a href="#explore-rrbs">DNA Methylation</a></li>
        <li><a href="#explore-chipseq">ChIP-seq</a></li>
        <li><a href="#explore-rnaseq">RNA-Seq</a></li>
    </ul>
</div>

<h2 id="explore-rrbs">DNA Methylation</h2>
<ul>
    <li><a href="#viewrrbs">View</a> data in JBR genome browser.</li>
    <li>Download preconfigured <a href="#sessionsrrbs">session files.</a></li>
    <li>Download <a href="#data">data</a> used in sessions.</li>
</ul>
<h3 id="viewrrbs">View</h3>
<p>
    Explore preconfigured online session in web <a
        href="https://github.com/JetBrains-Research/jbr">JBR Genome Browser</a>
    in one click: <a href="https://artyomovlab.wustl.edu/jbr/2018_rrbs_aging">
    https://artyomovlab.wustl.edu/jbr/2018_rrbs_aging</a>
</p>

<h3 id="sessionsrrbs">Sessions</h3>
<p>
    For those who prefer desktop genome browser applications, we prepared dedicated session files for the following
    genome browsers:
    <br>
    <a href="https://software.broadinstitute.org/software/igv/">IGV Genome Browser</a> and
    <a href="https://github.com/JetBrains-Research/jbr" target="_blank">JBR Genome
        Browser</a>.
    <br>
    Click on the corresponding link to download the application.
</p>
Preconfigured session file for <code>IGV</code> or <code>JBR</code> genome browsers:
<a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rrbs/2018_rrbs_aging.xml">DNA methylation session</a>.

<h2 id="explore-chipseq">ChIP-Seq</h2>
<ul>
    <li><a href="#view">View</a> data in the JBR and UCSC genome browsers.</li>
    <li>Download preconfigured <a href="#sessions">session files.</a></li>
    <li>Download <a href="#data">data</a> used in sessions.</li>
</ul>
<h3 id="view">View</h3>
<p>
    Explore preconfigured online session in web <a
        href="https://github.com/JetBrains-Research/jbr">JBR Genome Browser</a>
    in one click.
</p>
<table class="table table-bordered table-nonfluid">
    <thead class="thead-light">
    <tr>
        <th>Modification</th>
        <th class="text-center">Session</th>
    </tr>
    </thead>
    <tbody>
    <tr>
        <th>H3K27ac</th>
        <td class="text-center"><a href="https://artyomovlab.wustl.edu/jbr/2018_h3k27ac_aging">2018_h3k27ac_aging</a></td>
    </tr>
    <tr>
        <th>H3K27me3</th>
        <td class="text-center"><a href="https://artyomovlab.wustl.edu/jbr/2018_h3k27me3_aging">2018_h3k27me3_aging</a></td>
    </tr>
    <tr>
        <th>H3K36me3</th>
        <td class="text-center"><a href="https://artyomovlab.wustl.edu/jbr/2018_h3k36me3_aging">2018_h3k36me3_aging</a></td>
    </tr>
    <tr>
        <th>H3K4me1</th>
        <td class="text-center"><a href="https://artyomovlab.wustl.edu/jbr/2018_h3k4me1_aging">2018_h3k4me1_aging</a></td>
    </tr>
    <tr>
        <th>H3K4me3</th>
        <td class="text-center"><a href="https://artyomovlab.wustl.edu/jbr/2018_h3k4me3_aging">2018_h3k4me3_aging</a></td>
    </tr>
    </tbody>
</table>
<br>
<p>
    Explore preconfigured online sessions in the UCSC genome browser in one click.
</p>
<table class="table table-bordered table-nonfluid">
    <thead class="thead-light">
    <tr>
        <th>Modification</th>
        <th class="text-center">Basic</th>
        <th class="text-center">Extended</th>
    </tr>
    </thead>
    <tbody>
    @TABLE@
    </tbody>
</table>
Materials included in this table:
<ul>
    <li>
        Preconfigured sessions for the <a href="https://genome.ucsc.edu">UCSC Genome Browser</a>.
    </li>
    <li>
        Session Type - Basic or Extended.<br>
        <p>
            Basic session contains reads visualization in bigWig format, individual peaks for each sample,
            <br>
            and labels used in <a href="tools.html">semi-supervised peaks analysis</a>.
            <br>
            ENCODE tracks for classical CD14+ monocytes are included for control.
            <br>
            Extended session contains additional peaks produced by golden standard peak callers, i.e., <a
                href="https://github.com/taoliu/MACS">MACS2</a>
            and <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4152844/">SICER</a>.
        </p>
    </li>
</ul>


<h3 id="sessions">Sessions</h3>
<p>
    For those who prefer desktop genome browser applications, we prepared dedicated session files for the following
    genome browsers:
    <br>
    <a href="https://software.broadinstitute.org/software/igv/">IGV Genome Browser</a> and
    <a href="https://github.com/JetBrains-Research/jbr" target="_blank">JBR Genome
        Browser</a>.
    <br>
    Click on the corresponding link to download the application.
</p>
<p>
    See the section <a href="tools.html">Tools</a> to learn how <code>JBR</code> Genome Browser can be used for
    visual peak calling.
</p>
<table class="table table-bordered table-nonfluid">
    <thead class="thead-light">
    <tr>
        <th>Modification</th>
        <th class="text-center">Basic</th>
        <th class="text-center">Extended</th>
    </tr>
    </thead>
    <tbody>
    @TABLE2@
    </tbody>
</table>
Materials included in this table:
<ul>
    <li>
        Preconfigured session file for <code>IGV</code> or <code>JBR</code> genome browsers.
    </li>
    <li>
        Session types and content is described in the <a href="#explore">Explore</a> section.
    </li>
</ul>

<h2 id="explore-rnaseq">RNA-Seq</h2>
<p>
    Explore RNA-seq data in <a href="https://artyomovlab.wustl.edu/phantasus/phantasus-tutorial.html">Phantasus</a>
    - web-application for visual and interactive gene expression analysis.<br>
<ul>
    <li>
        <a href="https://artyomovlab.wustl.edu/phantasus/?preloaded=aging_paper_mono_raw">Raw counts</a>
    </li>
    <li>
        <a href="https://artyomovlab.wustl.edu/phantasus/?preloaded=aging_paper_mono_deseq">DESeq2 normlization</a>
    </li>
    <li>
        <a href="https://artyomovlab.wustl.edu/phantasus/?preloaded=aging_paper_mono_log2quant">Log2 & quantile
            normalization</a>
    </li>
</ul>

Explore RNA-Seq data for new experiment with monocytes differentiation and stimulation.
<ul>
    <li>
        <a href="https://artyomovlab.wustl.edu/phantasus/?preloaded=aging_paper_lps_macs_raw">Raw counts</a>
    </li>
    <li>
        <a href="https://artyomovlab.wustl.edu/phantasus/?preloaded=aging_paper_lps_macs_deseq">DESeq2
            normlization</a>
    </li>
    <li>
        <a href="https://artyomovlab.wustl.edu/phantasus/?preloaded=aging_paper_lps_macs_log2quant">Log2 & quantile
            normalization</a>
    </li>
</ul>

<h2 id="data">Data</h2>
All the data is available in <a href="download_data.html">Download Data</a> section.


