<!-- To be included inside template.html -->

<div class="page-header">
    <h1>Epigenetic aging of classical monocytes from healthy individuals</h1>
</div>
<p>This page contains description and materials for the experiment.</p>
<div>
    <ul>
        <li><a href="#experiment">Experiment</a> section contains summary information</li>
        <li><a href="methods.html">Methods</a> section contains a detailed description of experimental protocols and
            computational pipelines
        </li>
        <li><a href="tools.html">Tools</a> section describes a novel semi-supervised peak calling approach used to
            process
            the dataset
        </li>
        <li><a href="explore_data.html">Explore Data</a> section allows easy visualisation of generated datasets</li>
        <li><a href="download_data.html">Download Data</a> section provides links for download of the data</li>
    </ul>
</div>
<h2 id="experiment">Experiment</h2>
<p>The impact of healthy aging on molecular programming of immune cells is poorly understood. Here, we
    report comprehensive characterization of healthy aging in human monocytes, with a focus on
    transcriptomic and epigenomic alterations, as well as the corresponding proteomic and metabolomic data
    for plasma, using healthy cohorts of 20 young and 20 older individuals (~27 and ~64 years old on average).
    For each individual, we have performed RRBS-based DNA methylation profiling, which revealed a set of
    age-related differentially methylated regions (DMRs) that either gain or lose methylation with age.
    Optimized ultra-low-input ChIP-seq (ULI-ChIP-seq) data acquisition and analysis pipelines allowed us to
    link hypo- and hypermethylated DMRs to distinct chromatin modification patterns. Specifically,
    hypermethylation events were associated with H3K27me3 marked CpG islands, while hypomethylated
    DMRs were enriched in H3K4me1 marked regions. Further integrative analysis of the transcriptomic data
    established a connection between identified age-associated alterations in the DNA methylome and
    corresponding age-associated transcriptional changes.
</p>
<img src="experiment.png" style="height: 100%; width: 100%; object-fit: contain"/>
<br/>
<br/>
<h2>Analysis</h2>
<p>
    To our knowledge, the comparative epigenetic study of this scale has never been undertaken previously in the context
    of human aging. Since the peak calling routine presents a significant challenge in working with large scale human
    epigenetic data, we developed a novel semi-supervised peak calling approach applicable to datasets of this scale.
    <br>
    Along with a pioneer dataset, we provide a completely <a href="tools.html">new integrative solution</a>
    for ChIP-seq analysis.
</p>

<h2>Publication</h2>
<p>
    A preprint is available on bioRxiv: <a href="https://biorxiv.org/lookup/doi/10.1101/2020.05.10.087023">https://biorxiv.org/lookup/doi/10.1101/2020.05.10.087023</a>
</p>
